internal/dao/tests/pubkey_test.go

```diff
@@ -44,90 +74,136 @@ func teardownPubkey(_ *testing.T) {
-	require.NoError(t, err)
 
-	assert.Equal(t, pk, pk2)
+	t.Run("success", func(t *testing.T) {
+		pk := newLukas2013Key()
+		err := pkDao.Create(ctx, pk)
+		require.NoError(t, err)
+
+		pk2, err := pkDao.GetById(ctx, pk.ID)
+		require.NoError(t, err)
+		assert.Equal(t, pk, pk2)
+	})
+
+	t.Run("fingerprint generation", func(t *testing.T) {
+		pk := newLukas2011Key()
+		err := pkDao.Create(ctx, pk)
+		require.NoError(t, err)
+
+		assert.Equal(t, "SHA256:JHqC1mQ9yannTQ32QEi0JfOFbZSPZVq5O68rmpMh7Lo", pk.Fingerprint)
+	})
+
+	t.Run("validation error on name", func(t *testing.T) {
+		pk := newLukas2021Key()
+		pk.Name = ""
+		err := pkDao.Create(ctx, pk)
+		require.ErrorAs(t, err, &validator.ValidationErrors{})
+	})
+
+	t.Run("validation error on body", func(t *testing.T) {
+		pk := newLukas2021Key()
+		pk.Body = ""
+		err := pkDao.Create(ctx, pk)
+		require.ErrorAs(t, err, &validator.ValidationErrors{})
+	})
 }
 
-func TestCreatePubkeyFingerprintSuccess(t *testing.T) {
+func TestPubkeyList(t *testing.T) {
 	pkDao, ctx := setupPubkey(t)
 	defer teardownPubkey(t)
-	pk := createLukas2013Key()
-	err := pkDao.Create(ctx, pk)
-	require.NoError(t, err)
 
-	assert.Equal(t, "SHA256:ENShRe/0uDLSw9c+7tc9PxkD/p4blyB/DTgBSIyTAJY", pk.Fingerprint)
+	t.Run("success", func(t *testing.T) {
+		pubkeys, err := pkDao.List(ctx, 1, 0)
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(pubkeys))
+	})
+
+	t.Run("with offset", func(t *testing.T) {
+		newKey := newLukas2013Key()
+		err := pkDao.Create(ctx, newKey)
+		require.NoError(t, err)
+
+		pubkeys, err := pkDao.List(ctx, 1, 0)
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(pubkeys))
+
+		pubkeys, err = pkDao.List(ctx, 1, 1)
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(pubkeys))
+		require.Contains(t, pubkeys, newKey)
+	})
 }
 
-func TestCreatePubkeyFingerprintFailure(t *testing.T) {
+func TestPubkeyUpdate(t *testing.T) {
 	pkDao, ctx := setupPubkey(t)
 	defer teardownPubkey(t)
-	pk := createLukas2021Key()
-	err := pkDao.Create(ctx, pk)
-	assert.Error(t, err)
-}
 
-func TestListPubkey(t *testing.T) {
-	pkDao, ctx := setupPubkey(t)
-	defer teardownPubkey(t)
-	newKey := createLukas2013Key()
-	err := pkDao.Create(ctx, newKey)
-	require.NoError(t, err)
-	pubkeys, err := pkDao.List(ctx, 100, 0)
-	require.NoError(t, err)
-	assert.Equal(t, 2, len(pubkeys))
-	require.Contains(t, pubkeys, newKey)
+	t.Run("success", func(t *testing.T) {
+		newKey := newLukas2013Key()
+		err := pkDao.Create(ctx, newKey)
+		require.NoError(t, err)
+
+		updatePk := newAnna2021Key()
+		updatePk.ID = newKey.ID
+		err = pkDao.Update(ctx, updatePk)
+		require.NoError(t, err)
+
+		dbPk, err := pkDao.GetById(ctx, updatePk.ID)
+		require.NoError(t, err)
+		assert.Equal(t, updatePk, dbPk)
+	})
+
+	t.Run("no rows", func(t *testing.T) {
+		updatePk := newAnna2021Key()
+		updatePk.ID = math.MaxInt64
+		err := pkDao.Update(ctx, updatePk)
+		require.ErrorIs(t, err, dao.ErrAffectedMismatch)
+	})
+
+	// TODO validators
```

Comment:
This TODO is for me to add even more pubkey tests (validators).

I can add it either during rebase of this PR or later.